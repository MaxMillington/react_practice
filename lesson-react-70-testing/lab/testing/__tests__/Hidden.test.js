// Hidden.test.js
import React from 'react';
import renderer from 'react-test-renderer';

import Hidden from '../src/components/Hidden';

describe('Hidden -- ', () => {
    let component, comp;

    beforeEach(() => {
        component = renderer.create(
            <Hidden />
        );
        comp = component.toJSON();
    });

    it('should exist', () => {
        expect(comp).not.toBeNull();
        expect(comp).toMatchSnapshot();
    });

    it('should accept a click event', () => {
        // TODO manually trigger the callback
        // TODO re-rendering
        expect(comp).toMatchSnapshot();
    });

    it('should return to normal after two click events', () => {
        // TODO manually trigger the callback twice
        // TODO re-rendering
        expect(comp).toMatchSnapshot();
    });

    it('should find a p.block at init time', () => {
        expect(comp.children).toHaveLength(2);
        expect(comp.children[1].props.className).toEqual('block');
    });

    it('should find a p.none after one click', () => {
        comp.props.onClick();
        comp = component.toJSON();
        expect(comp.children).toHaveLength(2);
        expect(comp.children[1].props.className).toEqual('none');
    });

    it('should find a p.block after two clicks', () => {
        // TODO manually do a click and re-render
        // TODO manually do another click and re-render
        expect(comp.children[1].props.className).toEqual('block');
    });

});
