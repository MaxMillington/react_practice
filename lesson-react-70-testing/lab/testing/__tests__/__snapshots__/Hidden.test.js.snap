exports[`Hidden --  should accept a click event 1`] = `
<div
  onClick={[Function]}>
  <span>
    Click Here
  </span>
  <p
    className="block">
    Can you see me now?
  </p>
</div>
`;

exports[`Hidden --  should exist 1`] = `
<div
  onClick={[Function]}>
  <span>
    Click Here
  </span>
  <p
    className="block">
    Can you see me now?
  </p>
</div>
`;

exports[`Hidden --  should return to normal after two click events 1`] = `
<div
  onClick={[Function]}>
  <span>
    Click Here
  </span>
  <p
    className="block">
    Can you see me now?
  </p>
</div>
`;
